src/bin/cfsctl.rs

```diff
@@ -179,25 +251,88 @@ async fn main() -> Result<()> {
             OciCommand::MetaLayer { ref name } => {
                 oci::meta_layer(&repo, name, None)?;
             }
-            OciCommand::PrepareBoot { ref name, bootdir } => {
-                let output = bootdir.unwrap_or(PathBuf::from("/boot"));
-                oci::prepare_boot(&repo, name, None, &output)?;
+            OciCommand::PrepareBoot {
+                ref config_name,
+                ref config_verity,
+                ref bootdir,
+                ref entry_id,
+                ref cmdline,
+            } => {
+                let verity = verity_opt(config_verity)?;
+                let mut fs = oci::image::create_filesystem(&repo, config_name, verity.as_ref())?;
+                let entries = fs.transform_for_boot(&repo)?;
+                let id = fs.commit_image(&repo, None)?;
+
+                let Some(entry) = entries.into_iter().next() else {
+                    bail!("No boot entries!");
+                };
+
+                let cmdline_refs: Vec<&str> = cmdline.iter().map(String::as_str).collect();
+                composefs::bootloader::write_boot_simple(
+                    &repo,
+                    entry,
+                    &id,
+                    bootdir,
+                    entry_id.as_deref(),
+                    &cmdline_refs,
+                )?;
+
+                let state = args
+                    .repo
+                    .as_ref()
+                    .map(|p: &PathBuf| p.parent().unwrap())
+                    .unwrap_or(Path::new("/sysroot"))
+                    .join("state")
```

Comment:
Surprised there isn't a const for this somewhere

Reply:
For "state"?  I suppose we could add this.  To be honest, I consider this chunk of the code to be little more than an example...